pkg/googlesheets/datasource_test.go

```diff
@@ -109,3 +114,114 @@ func TestGoogleSheetsDatasource_QueryData(t *testing.T) {
 		assert.NotNil(t, err)
 	})
 }
+
+func TestGoogleSheetsMultiTenancy(t *testing.T) {
+	const (
+		tenantID1 = "abc123"
+		tenantID2 = "def456"
+		addr      = "127.0.0.1:8000"
+	)
+
+	var instances []instancemgmt.Instance
+	factoryInvocations := 0
+	factory := datasource.InstanceFactoryFunc(func(ctx context.Context, settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+		factoryInvocations++
+		i, err := NewDatasource(ctx, settings)
+		if err == nil {
+			instances = append(instances, i)
+		}
+		return i, err
+	})
+
+	tp, err := datasourcetest.Manage(factory, datasourcetest.ManageOpts{Address: addr})
+	require.NoError(t, err)
+	defer func() {
+		err = tp.Shutdown()
+		t.Log("plugin shutdown error", err)
```

Comment:
I need to fix this upstream too - but we should change this to:

```suggestion
		err = tp.Shutdown()
		if err != nil {
			t.Log("plugin shutdown error", err)
		}
```